### ISSUE TITLE
freeCodeCamp's JS Curriculum, How Can It Be Improved?

### ISSUE THREAD

**Issue by FrontEndWebDeveloping** [FrontEndWebDeveloping/New-Issue#1]
# freeCodeCamp's Js Curriculum: One Big Improvement
## My Experience
I really loved freeCodeCamp's HTML and CSS course. Then I started into their Java Script curriculum and it went really bad. I felt like they wanted me to learn the language by copying, so that when I got to my first independent project I didn't have a clue what to do.

I went and did more research. I found a book that taught me the _principles_ and I did much better. I still really like freeCodeCamp, but this is a **BIG** problem. Java Script is a fundamental language. If they don't have a good curriculum it may spell bad things for them, and it's certainly pretty hard on students.

## The Fix
freeCodeCamp must teach students the _principles_ behind JS. This will at least double the usefulness of their JS course and will likely increase their popularity as a path to learning to code. Their first challenge should be a whole lot simpler, and also, more businesslike, not a game. It should be designed to take the students through the foundation of JS, and not shoot the into the clouds right away. The instructions should have in depth explanations of the reasons why. Also, students need to have more time to practice what they learn. The lessons should progress more slowly so that students can retain the information.

## To Sum Up
freeCodeCamp, as a whole, is a great place and project based learning is the best way to learn. But it must be managed rightly, with _reasonable expectations_ for the students comprehension and retention abilities.

**Comment by lasjorg** [comment]
Here is the related forum thread for more context and feedback.

https://forum.freecodecamp.org/t/the-js-course-from-the-perspective-of-a-recent-student/673914
